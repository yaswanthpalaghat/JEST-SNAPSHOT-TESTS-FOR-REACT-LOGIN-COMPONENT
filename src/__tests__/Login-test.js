import React from "react";
import Enzyme, { shallow, mount, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Login from "../Login";
import renderer from "react-test-renderer";

Enzyme.configure({ adapter: new Adapter() });

describe("Login Component", () => {
  it("should render without throwing an error", () => {
    expect(shallow(<Login />).exists(<form className="login"></form>)).toBe(
      false
    );
  });
  it("renders a email input", () => {
    expect(shallow(<Login />).find("#email").length).toEqual(1);
  });
  it("renders a password input", () => {
    expect(shallow(<Login />).find("#password").length).toEqual(1);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("Email input", () => {
    it("should respond to change event and change the state of the Login Component", () => {
      const wrapper = shallow(<Login />);
      wrapper.find("#email").simulate("change", {
        target: { name: "email", value: "yash@gmail.com" }
      });
      expect(wrapper.state("email")).toEqual("yash@gmail.com");
    });
  });

  describe("Password input", () => {
    it("should respond to change event and change the state of the Login Component", () => {
      const wrapper = shallow(<Login />);
      wrapper
        .find("#password")
        .simulate("change", { target: { name: "password", value: "yash" } });
      expect(wrapper.state("password")).toEqual("yash");
    });
  });
});
