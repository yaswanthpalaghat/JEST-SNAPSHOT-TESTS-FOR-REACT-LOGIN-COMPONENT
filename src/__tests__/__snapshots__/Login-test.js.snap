// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<form
  className="login"
>
  <label>
    Username
  </label>
  <input
    id="email"
    name="email"
    onChange={[Function]}
    type="text"
  />
  <label>
    Password
  </label>
  <input
    id="password"
    name="password"
    onChange={[Function]}
    type="password"
  />
  <button>
    Submit
  </button>
</form>
`;
